### ISSUE TITLE
Can not load image from 'http://localhost:8080/var/containers/Bundle' IONIC 3

### ISSUE THREAD

**Issue by elchory96** [mapsplugin/cordova-plugin-googlemaps#2558]
**I'm submitting a ...**  (check one with "x")
- [ ] question
- [x] any problem or bug report

**OS:**  (check one with "x")
- [ ] **Android**
- [x] **iOS**
- [ ] **Browser**

**cordova information:**  (run `$> cordova plugin list`) 

```
insert the output from the command here
```

**If you use `@ionic-native/google-maps`, please tell the package.json (only `@ionic-native/core` and `@ionic-native/google-maps` are fine mostly)**
```
@ionic-native/core : 
@ionic-native/google-maps : 
```

**Current behavior:**

Hello, what happens is that in my main marker of the map I have it calling an asset image that I have saved it, likewise the images for the MarkerCluster, but it tells me the following error:

> 2019-02-08 16:39:15.744866-0500 Simi Comercial[406:43574] ERROR: map_0_293471758616 Can not load image from 'http://localhost:8080/var/containers/Bundle/Application/272E987D-B4BB-43BD-B685-B079457B4193/Simi%20Comercial.app/www/assets/imgs/iconoappsimiAzul.png'.

I am compiling the app in tests on the mobile device. but I'm not using Livereload.

I work with IONIC 3.

**Expected behavior:**

Here are the screenshots

![captura de pantalla 2019-02-08 a la s 4 44 02 p m](https://user-images.githubusercontent.com/18226156/52507980-215df900-2bc1-11e9-933e-8e8ce24eaa60.png)

![captura de pantalla 2019-02-08 a la s 4 44 35 p m](https://user-images.githubusercontent.com/18226156/52507982-215df900-2bc1-11e9-9548-8ae13bf8ac1d.png)

![captura de pantalla 2019-02-08 a la s 4 45 03 p m](https://user-images.githubusercontent.com/18226156/52507983-215df900-2bc1-11e9-847a-adecd975cfd7.png)

![captura de pantalla 2019-02-08 a la s 4 45 22 p m](https://user-images.githubusercontent.com/18226156/52507984-215df900-2bc1-11e9-8e70-797c56466839.png)

**Screen capture or video record:**

**Related code, data or error log (please format your code or data):**

```
Please share your project files on Github or others(Bitbucket, Gitlabs...etc).
If you don't want to share your project files, please create a demo project, then share it.

Screen captures, and/or native logs(such as Logcat, xcode logs) are appreciate.

Giving much information, you are waiting time is less.
Thank you for your cooperation.
```

**Support this plugin activity**

I appreicate if you give me a beer :beer: from [here](
https://www.paypal.com/cgi-bin/webscr?cmd=_donations&business=SQPLZJ672HJ9N&lc=US&item_name=Cordova%2dGoogleMaps%2dPlugin&amount=5%2e00&currency_code=USD&bn=PP%2dDonationsBF%3abtn_donate_LG%2egif%3aNonHosted)

**Comment by wf9a5m75** [comment]
Please share your project files on github repository.

**Comment by elchory96** [comment]
I already shared the repository in github, the map is in the module "Properties"

**Comment by wf9a5m75** [comment]
For private repo support, I appreciate if you donate $5 to the project for maps plugin activity.

**Comment by elchory96** [comment]
Ok, I understand, I will talk to the company to make the respective donation, but would you help me with that problem in ios for now?

**Comment by elchory96** [comment]
Hello, sorry for the insistence, is that I need to upload the application to the App Store but with that error I can not upload it.

Do you have any news on this subject?

**Comment by wf9a5m75** [comment]
My private time is very limited. Thus, free support is low priority.
I tried to rebuild your project files, but App does not start after splash screen.

![screen shot 2019-02-17 at 10 11 37 pm](https://user-images.githubusercontent.com/167831/52931402-0ac24b00-3301-11e9-8d51-c5e7c35bce33.png)

Regarding the error messages, you need to update JS code, especially Firebase around.

<img width="1326" alt="screen shot 2019-02-17 at 10 09 43 pm" src="https://user-images.githubusercontent.com/167831/52931314-cb93fa00-3300-11e9-9ea9-71e98573a1ea.png">
<img width="1326" alt="screen shot 2019-02-17 at 10 09 48 pm" src="https://user-images.githubusercontent.com/167831/52931315-cb93fa00-3300-11e9-9c3e-fc6c639b318a.png">
<img width="1326" alt="screen shot 2019-02-17 at 10 09 51 pm" src="https://user-images.githubusercontent.com/167831/52931316-cc2c9080-3300-11e9-9aae-2e7df61bae1b.png">

**Comment by wf9a5m75** [comment]
ping

**Comment by wf9a5m75** [comment]
Since there is no response for a while, I close this thread.

**Comment by JamesPoel** [comment]
Did you find a solution? I also cannot load cluster images.

**Comment by UlisesCeca** [comment]
did you find anything @JamesPoel  ?

I am getting error only with some markers' icons.

Weird thing is that other markers use the same icons and work

**Comment by JamesPoel** [comment]
> did you find anything @JamesPoel ?
> 
> I am getting error only with some markers' icons.
> 
> Weird thing is that other markers use the same icons and work

I'm afraid not! We're still battling with it. Sometimes the markers load, other times they fail to load with the "Can not load image from..." error. It's very frustrating!

**Comment by UlisesCeca** [comment]
I fixed it, I noticed I had my apikey for android and ios switched. So my ios apikey was wrong.

**Comment by JamesPoel** [comment]
> I fixed it, I noticed I had my apikey for android and ios switched. So my ios apikey was wrong.
> 
> Can you see the streets properly? If you can't see them, this is the problem

Hmm yess.. I can see the streets and my api keys are correct (they're actually identical on both android and ios for me).

I have worked round our problem by using package https://ionicframework.com/docs/v3/native/base64/ to base64 encode the icon url's and then setting the url property to the base64 instead of the url, which works fine. It's a bit hacky but at least my icons show up now. I'm not sure what the original problem is!

**Comment by paulnagle** [comment]
I am seeing the same issue on Ionic 4.. I get the error messages about not being able to load the marker and marker cluster images when I first launch the map. As I move around the map a bot, the images start to appear sporadically.. I dunno, it's a weird one @JamesPoel Would you mind posting an example of the code that you used to get around this? Do you actually base64 the marker image, or just the url to the image??

**Comment by JamesPoel** [comment]
@paulnagle this was my exact issue. Most icon/cluster images failing to load with an error message until you pinch/scroll around for a bit, then the start to appear/disappear at random before all working fine after a minute or so of scrolling around. 

I base64'd the actual file, not the url (should have clarified). The package I found (linked in my previous comment) contains a `this.base64.encodeFile(...)` function which takes a local/absolute file path as an argument and returns the base64 data in the form of a promise. 

I don't have the exact code I used to hand (I'm on my mobile at the moment) but the gist of it was:

```
let filePath = 'assets/images/icon.png';
this.base64.encodeFile(filePath).then((base64: string) => {
   this.map.addMarker( { icon : base64, ...} );
}
```
This stopped the problem completely. It also worked on my cluster markers.
